### PR TITLE
Added support for agreements

### PR DESCRIPTION
**Pull request by camilloop**
This feature adds support for Magento agreements (Stores -> Terms and Conditions).

### CODE REVIEW

File: src/module-vsbridge-indexer-agreement/Index/Mapping/Agreement.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            'content_height' => ['type' => FieldInterface::TYPE_TEXT],
+            'checkbox_text' => ['type' => FieldInterface::TYPE_TEXT],
+            'is_html' => ['type' => FieldInterface::TYPE_BOOLEAN],
+            'mode' => ['type' => FieldInterface::TYPE_INTEGER],
+        ];
+
+        $mappingObject = new DataObject();
+        $mappingObject->setData('properties', $properties);
+
+        return $mappingObject->getData();
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-agreement/Index/Mapping/Agreement.php:34]: we really don't need this, you can just do `return $properties`

File: src/module-vsbridge-indexer-agreement/Model/Indexer/Action/Agreement.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    private $resourceModel;
+
+    /**
+     * Agreement constructor.
+     *
+     * @param AgreementResource $agreementResource
+     */
+    public function __construct(
+        AgreementResource $agreementResource
+    ) {
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-agreement/Model/Indexer/Action/Agreement.php:26]: we can have it one line `public function __construct(        AgreementResource $agreementResource    ) `

File: src/module-vsbridge-indexer-agreement/Plugin/Indexer/Agreement/Save/UpdateAgreement.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        $this->resourceModelAgreement = $resourceModelAgreement;
+    }
+
+    /**
+     * @param Agreement $agreement
+     * @param Agreement $result
+     *
+     * @return Agreement
+     */
+    public function afterAfterSave(Agreement $agreement, Agreement $result)
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-agreement/Plugin/Indexer/Agreement/Save/UpdateAgreement.php:44]: we can probably use the return type declaration in this case: `afterAfterSave(Agreement $agreement, Agreement $result): Agreement`

same for `afterAfterDeleteCommit` function

File: src/module-vsbridge-indexer-agreement/etc/module.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
+<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Module/etc/module.xsd">
+    <module name="Divante_VsbridgeIndexerAgreement" setup_version="1.0.0">
+        <sequence>
+            <Divante_VsbridgeIndexerCore/>
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-agreement/etc/module.xml:5]: We should also add dependency to Magento core module `Magento_CheckoutAgreements`

File: src/module-vsbridge-indexer-agreement/Index/Mapping/Agreement.php (hunk 1 above)
Comment (camilloop) [reply src/module-vsbridge-indexer-agreement/Index/Mapping/Agreement.php]: fixed

File: src/module-vsbridge-indexer-agreement/Model/Indexer/Action/Agreement.php (hunk 2 above)
Comment (camilloop) [reply src/module-vsbridge-indexer-agreement/Model/Indexer/Action/Agreement.php]: fixed

File: src/module-vsbridge-indexer-agreement/Plugin/Indexer/Agreement/Save/UpdateAgreement.php (hunk 3 above)
Comment (camilloop) [reply src/module-vsbridge-indexer-agreement/Plugin/Indexer/Agreement/Save/UpdateAgreement.php]: fixed

File: src/module-vsbridge-indexer-agreement/etc/module.xml (hunk 4 above)
Comment (camilloop) [reply src/module-vsbridge-indexer-agreement/etc/module.xml:5]: fixed

File: src/module-vsbridge-indexer-agreement/Index/Mapping/Agreement.php [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    /**
+     * @return array|mixed|null
+     */
+    public function getMappingProperties()
+    {
+        $properties = [
+            'id' => ['type' => FieldInterface::TYPE_LONG],
+            'content' => ['type' => FieldInterface::TYPE_TEXT],
+            'active' => ['type' => FieldInterface::TYPE_BOOLEAN],
+            'name' => ['type' => FieldInterface::TYPE_TEXT],
```
Comment (afirlejczyk) [inline src/module-vsbridge-indexer-agreement/Index/Mapping/Agreement.php:23]: what do you think of switching 'name' to 'title'?

Comment (camilloop) [reply src/module-vsbridge-indexer-agreement/Index/Mapping/Agreement.php:23]: My way of thinking is that it should be the same name as in the magento database, but if you feel that 'title' is better in this case I can change it. Please confirm that I should change it

Comment (camilloop) [reply src/module-vsbridge-indexer-agreement/Index/Mapping/Agreement.php]: just like we talked on priv, I changed to 'title'

**Pull request merged**